cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -116,18 +122,45 @@ private void addPrefixes(@Nonnull Map<String, String> prefixes,
   }
 
   private void addTypes(Configuration configuration, RdfStoreService rdfStoreService, Model shacl) {
-    List<String> types = getTypes(rdfStoreService);
+    List<String> types = getTypes(configuration, rdfStoreService);
     log.debug("(addTypes) found {} types", types.size());
     types.forEach(type -> addType(configuration, rdfStoreService, shacl, type));
   }
 
-  private List<String> getTypes(RdfStoreService rdfStoreService) {
+  private List<String> getTypes(Configuration configuration, RdfStoreService rdfStoreService) {
     List<Map<String, RDFNode>> rows = getRows(rdfStoreService, "select-types.sparql");
     List<String> types = paginatedQuery.convertSingleColumnUriToStringList(rows);
-    Collections.sort(types);
+
+    types.sort(getIriComparator(configuration));
+
     return types;
   }
 
+  private Comparator<String> getIriComparator(Configuration configuration) {
+    return (t1, t2) -> {
+      int priorityT1 = configuration.getPriorityIndex(getNamespaceIri(t1));
+      int priorityT2 = configuration.getPriorityIndex(getNamespaceIri(t2));
+
+      if (priorityT1 >= 0 && priorityT2 >= 0) {
```

Comment:
relying on indexOf returning -1... there is something I have not seen in a long time :) 
(no need to change)